test/reply-earlyHints.test.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const Fastify = require('..')
+const { test } = require('tap')
+const http = require('http')
+const http2 = require('http2')
+
+const testResBody = 'Hello, world!'
+
+test('sends early hints', (t) => {
+  t.plan(6)
+
+  const fastify = Fastify({
+    logger: false
+  })
+
+  fastify.get('/', async (request, reply) => {
+    reply.writeEarlyHints({
+      link: '</styles.css>; rel=preload; as=style'
+    }, () => {
+      t.pass('callback called')
+    })
```

Comment:
Shall we offer a promisified version in case implementers wants to implement a log that depends on `early-hint` being written?

Reply:
I'm definitely not opposed to it

I wonder why node hasn't already done it

Reply:
Most of the Req/Res APIs are event based, so maybe to keep the line?

Reply:
By nature `Early Hints` should be fire and forget.
Nobody should rely on the existence of `early hints` in their application logic.

Reply:
See what you mean. My point was mostly to see if we could provide good ergonomics to different use cases.
Specially as early hints can be written more than once, and possibly links between hints being written (not necessarily acknowledge/fetched) can be needed/done.

But maybe is something worth to explore once it lands and see if there's a true need for it